### ISSUE TITLE
Better handling of file encodings

### ISSUE THREAD

**Issue by lwhite1** [jtablesaw/tablesaw#638]
We need to be able to specify an encoding/charset when opening a CSV or other text file

We _should_ support encoding detection via ICU4J (or better, if there is something better)

**Comment by benmccann** [comment]
You can specify. Pass in an `InputStreamReader` which allows specifying the charset:

https://docs.oracle.com/javase/7/docs/api/java/io/InputStreamReader.html#InputStreamReader(java.io.InputStream,%20java.nio.charset.Charset)

**Comment by lwhite1** [comment]
yes, but the process of figuring out what was wrong, what the correct
encoding was, and how to make Tablesaw do the right thing- took me an hour.
To learn how to change the encoding I read CsvReadOptions, then
ReadOptions, then grepped the code. That led me to the Source class, which
led me to the Builders with use streams (It doesn't say why streams have a
source, but files don't. Then I created a reader on the file and passed in
the encoding (which I figured out using ICU4J) and, *voilà, *I had the file
loaded with accents instead of question marks.

Numbers, Excel, and Pandas just did it correctly on the first try.

**Comment by lwhite1** [comment]
The implementation in #638 handles the most common case: using either of these

```
Table.read().csv("filename") 
// or 
Table.read().csv(file)
```
